src/dao/WavesDAO.js

```diff
@@ -0,0 +1,253 @@
+  }
+
+  hasBalancesStream () {
+    // Balance should not be fetched after transfer notification,
+    // it will be updated from the balances event stream
+    return true
+  }
+
+  getDecimals () {
+    return this._decimals
+  }
+
+  async getAccountBalances () {
+    return await this._wavesProvider.getAccountBalances(this._asset)
+  }
+
+  async getAccountBalance () {
+    return await this.getAccountBalances()
+  }
+
+  accept (transfer: TransferExecModel) {
+    setImmediate(() => {
+      this.emit('accept', transfer)
+    })
+  }
+
+  reject (transfer: TransferExecModel) {
+    setImmediate(() => {
+      this.emit('reject', transfer)
+    })
+  }
+
+  // TODO @ipavlenko: Replace with 'immediateTransfer' after all token DAOs will start using 'submit' method
+  transfer (from: string, to: string, amount: BigNumber, token: TokenModel, feeMultiplier: Number = 1) {
+    this.submit(from, to, amount, token, feeMultiplier)
+  }
+
+  submit (from: string, to: string, amount: BigNumber, token: TokenModel, feeMultiplier: Number = 1) {
+    setImmediate(async () => {
+      this.emit('submit', new TransferExecModel({
+        title: `tx.Waves.${this._asset ? this._asset : 'WAVES'}.transfer.title`,
+        from,
+        to,
+        amount: new Amount(amount, token.symbol()),
+        amountToken: token,
+      }))
+    })
+  }
+
+  // TODO @ipavlenko: Rename to 'transfer' after all token DAOs will start using 'submit' method and 'trans'
+  async immediateTransfer (from: string, to: string, amount: BigNumber, token: TokenModel, feeMultiplier: Number = 1) {
+    try {
+      return await this._wavesProvider.transfer(from, to, amount, this._asset)
+    } catch (e) {
+      // eslint-disable-next-line
+      console.log('Transfer failed', e)
+      throw e
+    }
+  }
+
+  async getTransfer (id, account, skip, offset): Promise<Array<TxModel>> {
+    const txs = []
+    try {
+      const txsResult = await this._wavesProvider.getTransactionsList(account, id, skip, offset)
+      for (const tx of txsResult) {
+        if (tx.value > 0) {
+          txs.push(new TxModel({
+            txHash: tx.txHash,
+            blockHash: tx.blockHash,
+            blockNumber: tx.blockNumber,
+            time: tx.time,
+            from: tx.from,
+            to: tx.to,
+            symbol: this._symbol,
+            value: new Amount(tx.value, this._symbol),
+            fee: new Amount(tx.fee, this._symbol),
+            blockchain: BLOCKCHAIN_WAVES,
+          }))
+        }
+      }
+    } catch (e) {
+      // eslint-disable-next-line
+      console.warn('WavesDAO getTransfer', e)
+    }
+    return txs
+  }
+
+  watch (/*account*/): Promise {
+    return Promise.all([
+      this.watchTransfer(),
+      this.watchBalance(),
+    ])
+  }
+
+  async watchTransfer () {
+    this._wavesProvider.addListener(EVENT_TX, async ({ tx }) => {
+      if (tx.unconfirmed) {
+        if (!this._mosaic) {
+          if (!tx.mosaics) {
```

Comment:
Mosaics or Assets?

Mosaics for NEM,
Assets for Waves

It seems the code related to Assets is not fully ported